shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h

```diff
@@ -287,8 +293,38 @@ class FlutterPlatformViewsController {
   // Pushes the view id of a visted platform view to the list of visied platform views.
   void PushVisitedPlatformView(int64_t view_id) { visited_platform_views_.push_back(view_id); }
 
+  // Visible for testing.
+  void CompositeWithParams(int64_t view_id, const EmbeddedViewParams& params);
+
+  const EmbeddedViewParams& GetCompositionParams(int64_t view_id) const {
+    return current_composition_params_.find(view_id)->second;
+  }
+
  private:
-  using LayersMap = std::map<int64_t, std::vector<std::shared_ptr<FlutterPlatformViewLayer>>>;
+  struct LayerData {
+    SkRect rect;
+    int64_t view_id;
+    int64_t overlay_id;
+    std::shared_ptr<FlutterPlatformViewLayer> layer;
+  };
+
+  using LayersMap = std::map<int64_t, std::vector<LayerData>>;
+
+  /// Update the buffers and mutate the platform views in CATransaction on the platform thread.
+  void PerformSubmit(const LayersMap& platform_view_layers,
+                     const std::vector<SurfaceFrame::DeferredSubmit>& callbacks,
+                     std::map<int64_t, EmbeddedViewParams>& current_composition_params,
+                     const std::unordered_set<int64_t>& views_to_recomposite,
+                     const std::vector<int64_t>& composition_order,
+                     const std::vector<std::shared_ptr<FlutterPlatformViewLayer>>& unused_layers,
+                     const std::vector<UIView*>& views_to_dispose);
+
+  /// @brief Populate any missing overlay layers.
+  ///
+  /// This requires posting a task to the platform thread and blocking on its completion.
+  void CreateMissingOverlays(GrDirectContext* gr_context,
```

Comment:
I wonder if the map of layers can contain a nullopt for layers that should go into a specific spot but haven't been constructed yet. Then, in the call that currently reorders, you could inflate them to create the actual view before inserting it in the right spot. Just a suggestion.

Reply:
That is another approach we can do, but if we defer creating the overlay layers we also need to defer constructing encoding the raster work.

Reply:
longer term, we should be able to construct the backing layer on any thread (once we fully move away from CAMetalLayer), and so we could remove this hack